api/server/backend/build/backend.go

```diff
@@ -72,8 +72,11 @@ func (b *Backend) Build(ctx context.Context, config backend.BuildConfig) (string
 		if imageID, err = squashBuild(build, b.imageComponent); err != nil {
 			return "", err
 		}
+	}
+
+	if options.Squash || useBuildKit {
 		if config.ProgressWriter.AuxFormatter != nil {
-			if err = config.ProgressWriter.AuxFormatter.Emit(types.BuildResult{ID: imageID}); err != nil {
+			if err = config.ProgressWriter.AuxFormatter.Emit("", types.BuildResult{ID: imageID}); err != nil {
```

Comment:
Why no ID in here, and in `emitImageID`

Reply:
@tonistiigi because that was the previous behavior: anything that used Emit, did not set the ID field.